scripts/emerge/ingest_to_tdr.py

```diff
@@ -21,84 +21,116 @@ def get_access_token():
 
 
-def call_ingest_dataset(control_file_path, target_table_name, dataset_id):
+def create_ingest_dataset_request(control_file_path, target_table_name, load_tag=None):
+    """Create the ingestDataset request body."""
+
+    load_dict = {"format": "json",
+                 "path": control_file_path,
+                 "table": target_table_name,
+                 "resolve_existing_files": "true",
+                 "updateStrategy": "replace"
+                }
+    # if user provides a load_tag, add it to request body
+    if load_tag:
+        load_dict["load_tag"] = load_tag
+
+    load_json = json.dumps(load_dict) # dict -> json
+
+    return load_json
+
+
+def call_ingest_dataset(control_file_path, target_table_name, dataset_id, load_tag=None):
     """Create the ingestDataset API json request body and call API."""
 
-    load_json = json.dumps({"format": "json",
-                            "path": control_file_path,
-                            "table": target_table_name,
-                            "resolve_existing_files": "true",
-                            "updateStrategy": "replace"
-                            })
-    load_job_response = load_data(dataset_id, load_json)
-
-    print(f"ingestDataset request body: \n {load_json} \n")
-    print(f"ingestDataset response body: \n {load_job_response} \n")
-
-    ingestDataset_job_id = load_job_response["id"]
-    ingestDataset_status_code = load_job_response["status_code"]
-
-    if ingestDataset_status_code != 202:
-        print("file ingest to TDR dataset failed.")
-        print(f"please refer to error message and retry ingest.")
-        print(f"ingestDataset response body: \n {load_job_response} \n")
-        raise ValueError(load_job_response)
-
-    # if ingestDataset_status_code == 202:  # if ingestDataset starts running successfully
-    job_status_code, job_response = get_job_status_and_result(ingestDataset_job_id)
-    while job_status_code == 202:           # while job is still running
-        time.sleep(10)  # wait 10 seconds
-        job_status_code, job_response = get_job_status_and_result(ingestDataset_job_id)
-
-    if job_status_code != 200:              # when job completes but not success
-        error_message = job_response["errorDetail"]
-        print(f"Load job finished but did not succeed: {error_message}")
-        raise ValueError(error_message)
+    ingest_dataset_request = create_ingest_dataset_request(control_file_path, target_table_name, load_tag) # create request for ingestDataset
+    print(f"ingestDataset request body: \n {ingest_dataset_request} \n")
+
+    ingest_response = ingest_dataset(dataset_id, ingest_dataset_request) # call ingestDataset
+    print(f"ingestDataset response body: \n {ingest_response} \n")
+
+    ingest_job_id = ingest_response["id"] # check for presence of id in ingest_dataset()
+    ingest_status_code = ingest_response["status_code"]
+
+    if ingest_status_code != 202: # job submitted but fails
```

Comment:
didn't you already check for this inside `ingest_dataset` ? (line 68)

i think you can delete these 3 lines

Reply:
ah good catch.